ocs_ci/ocs/utils.py

```diff
@@ -975,8 +981,11 @@ def collect_ocs_logs(dir_name, ocp=True, ocs=True, mcg=False, status_failure=Tru
             os.path.expanduser(ocsci_config.RUN["log_dir"]),
             f"{dir_name}_{ocsci_config.RUN['run_id']}",
         )
-
-    if ocs:
+    # Restricting to one time log collection for both OCS&OCP in case of status failure
+    must_gather_count = ocsci_config.REPORTING.get("must_gather_count")
+    skip_ocs_collection = must_gather_count.get("ocs") >= 1 and not status_failure
```

Comment:
```suggestion
    collect_logs_on_success_run = config.REPORTING.get("collect_logs_on_success_run")
    skip_ocs_collection = (must_gather_count.get("ocs") >= 1 and collect_logs_on_success_run) and not status_failure
```

I think something like this for this and line below will fix it as well